### DISCUSSION TITLE
Strange things happening ...

### DISCUSSION THREAD

**Discussion by Woisek** [vladmandic/sdnext#91] [Q&A]
First, I want to thank @vladmandic for this fork. Besides some minor problems when installing, it feels a lot snappier then A1111.
But unfortunately, there are some things, that are irritating, at least for me.

1. The inbuild extensions are not there:
sd-dynamic-thresholding
sd-extension-aesthetic-scorer
sd-extension-steps-animation
sd-extension-system-info
sd-webui-controlnet
sd-webui-model-converter
seed_travel
stable-diffusion-webui-images-browser
stable-diffusion-webui-rembg
Why and how can they be installed/activated?

2. When doing something that needs to refresh/reload the UI (like changing the settings) the refresh/reload does nothing. After installing an extension, a reload does nothing, the extensions is not showing up. One has to restart the WebUI completely, which makes a refresh/reload nonsense.

3. The default settings for prompt and sampler are nice but ... where can I change/delete them?? I understand that it is well-intentioned, but it is still not helpful.

4. A Clip Skip slider is besides the CFG slider, which I don't want to have there. Where can this be removed or configured not to show up?

That's the most obvious for me for now. I hope you can help to solve this, please.

**Comment by vladmandic** [comment]
re: built-in extensions

currently new launcher/installer `automatic.sh` is linux-only and it is responsible for initializing submodules.
if you're on a different platform (windows or mac), wait a day or two, new launcher is on the way.

re: when gradio queues are enabled, reload is simply broken as there is no way to restart gradio in-place.
(old reload was basically killing server and restarting it behind the scenes, nothing fancier)
until there is a way, i'll remove reload buttons as they are misleading.

re: defaults are in `config.json` and `ui-config.json`

re: clip-skip - sorry, not going to remove it just like that. there are a lot of options that some users don't use, what should i do about that?

**Reply by Woisek** [reply]
OK, thanks for the info, this makes sense now.

> re: clip-skip - sorry, not going to remove it just like that. there are a lot of options that some users don't use, what should i do about that?

Well, CLIP-Skip can be put into the quick-Settings if someone really always need it between each generations. That's why I thought, it doesn't have to be "fixed" where it is right now.
It's no big deal, I can hide it myself via CSS, but it's not "clean". :)

**Comment by Woisek** [comment]
Just a note here @ ui-config,json:
I noticed, there are multiple duplicates in there, mainly only the upper and lower case is different. Is this intentional?

**Comment by vladmandic** [comment]
`ui-config.json` is auto-populated by current settings of ui components. as webui evolved over time, some components were renamed and you end up with old and new settings. need to spend some time to clean it up completely.

**Reply by Woisek** [reply]
Ah, OK. Thanks for clearification. 👍

**Reply by vladmandic** [reply]
i've recreated `ui-config.json` from scratch in the latest version, should not have any obsolete entries anymore.

**Comment by Woisek** [comment]
Cool stuff! Thanks!
An urgent feedback right away: Please **ask** the user if he wants to download the v1.5 checkpoint or not. I'm sitting here, waiting for a completely useless 4GB unbelievable slow download to finish before the install is continued. 😶

**Reply by vladmandic** [reply]
i agree. current behavior is default in original a1111, but i'll change it.

**Reply by Woisek** [reply]
Oh, and while you at it, please correct the circumstance to not check after a finished install. I noticed, after a restart, the donwload started again, because my v1.5 model is named different and apparently the system thinks the file does not exist and wants to download it again.
THANKS! 😊

**Reply by vladmandic** [reply]
i'll check, but that should work - my fork should try to load model as specified in configuration and if that is not available, load first available on from models folder and only if none are available, start a download of a default model (which i need to change). but i'll be tinkering with that code quite a lot soon as i want to support multiple folders for models.

**Reply by Woisek** [reply]
> i want to support multiple folders for models.
Oh, that would be not bad, as I sort/order my models in custom folders ... 🤪

**Reply by Woisek** [reply]
I see, defined temp folders are still not created when not existent ... 🤨
And a complete restart of WebUI is still necessary after installing an extension or other stuff ...

**Comment by vladmandic** [comment]
re: temp folders, i'll take a look, i haven't paid attention to that at all. care to open an issue with details?
re: server restart, i'm affraid thats a bit more difficult as gradio does not have shutdown option so to cleanly restart just the ui without the need to restart entire server. i'll take a second look if i can hack something, really the question is how to stop gradio once its running, otherwise there is no way to reload ui components.

**Reply by Woisek** [reply]
> care to open an issue with details?

I don't know how often I did this ... 🙄
Is it OK if I link to the last one I did? -> https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/9374#issuecomment-1507518186

@ server restart: I hate to say it, but: A1111 does it. 😐
And something completely different: Where is the Checkpoint Merger ... ?? 😶

**Reply by vladmandic** [reply]
a) yes.
b) not anymore, gradio queues break that and having queues is for me more important than restart
c) i've moved checkpoint merger under train tab as number of top-level tabs was getting a bit high.
although there is one bug i need to fix in checkpoint merger, its trying to read hardcoded cmdflag that doesn't exist anymore. might still work, i just noticed it.

**Reply by Woisek** [reply]
a) OK, perfect
b) Hmm ... I use the last available version of A1111 with the new Gradio, but clicking "Apply and restart UI" does indeed reload everything and previously installed extension show up. No need to shut down and restart WebUI ... 😌
c) What a relieve! I just started with checkpoint mergeing and panicked easily ... 😁

Thanks for you effort so far, great work! 👍

**Reply by vladmandic** [reply]
re: b - then you probably don't have gradio queues enabled. and queues are really important if you want to use server via api and not crash it if you send two requests at the same time.

**Reply by Woisek** [reply]
b) I haven't changed anything at all. When I open two tabs with A1111 and start a render in the first tab and one in the other, the second says "Waiting ..." then "In queue".
Did a re-check right now. 🧐

**Reply by vladmandic** [reply]
i'll add support for clean server ui restart to my backlog. how it was done in a1111 was a nightmare, sooo many issues behind the scenes. i know normal user may not care, but restarting ui would lead to a lot of internally inconsistent states.

**Reply by Woisek** [reply]
I hope you find a way. Till now everything is working great. 👍 It will get better. 😎
